app/controllers/redirections_controller.rb

```diff
@@ -73,33 +72,33 @@ def tagged_log(message)
   end
 
   def ensure_referrer_is_not_localhost
-    if HostValidator.new(referrer).invalid?
-      redirect_to page_path("localhost")
+    if creating_new_redirection?
+      if HostValidator.new(referrer).invalid?
+        redirect_to page_path("localhost")
+      end
     end
   end
 
   def ensure_referrer_is_not_blocked
-    if Redirection.exists?(slug: params[:slug])
-      # If the redirection already exists, let it through. We just want to
-      # prevent further redirections from being created.
-      return
```

Comment:
I replaced this block with `if creating_new_redirection?`, which does the same thing.